docs/oauth.md

```diff
@@ -13,46 +35,64 @@ To enable OAuth for your connector add an `<oauth-config>` field in the manifest
   manifest.xml
 
   <?xml version='1.0' encoding='utf-8' ?>
-
   <connector-plugin class=...>
     ...
     <dialect file='dialect.tdd'/>
-    <oauth-config file='oauthConfig.xml'/>    <!-- add this to to define your OAuth Configs -->
+    <oauth-config file='oauthConfigPing.xml'/>    <!-- add this to to define your OAuth Configs -->
   </connector-plugin>
 ```
 
-In your connectionFields.xml file make sure add an `authentication` field with a value equal to `oauth`.
+Starting in Tableau 2023.1, you can add multiple OAuth configs, embedded in the plugin. The end users may also provide external/custom OAuth configurations:
+- By installing the config files in the Tableau directory. See [External OAuth Config on Desktop](#externalcustom-oauth-configs-on-desktop)
+- By uploading the config for a site level OAuth client. See [Create Site OAuth Client](#create-site-oauth-client-20231) 
+
+However in both cases, at least one embedded config is still required.
 
 ```xml
-  connectionFields.xml Single Auth Type:
+  manifest.xml
+
+  <?xml version='1.0' encoding='utf-8' ?>
 
-  <field name="authentication" label="Authentication" category="authentication" value-type="string" editable="false" default-value="oauth" />
+  <connector-plugin class=...>
+    ...
+    <dialect file='dialect.tdd'/>
+    <oauth-config file='oauthConfigPing.xml'/> 
+    <oauth-config file='oauthConfigOkta.xml'/>
+  </connector-plugin>
 ```
 
+---
+In your *connectionFields.xml* file make sure to add an `authentication` field with a value equal to `oauth`. For example:
+
+## OAuth Only
+
 ```xml
-  connectionFields.xml Multiple Auth Types:
-
-  <field name="authentication" label="Authentication" category="authentication" value-type="selection" default-value="auth-user-pass" >
-    <selection-group>
-      <option value="auth-user-pass" label="Username and Password"/>
-      <option value="oauth" label="OAuth"/>
-    </selection-group>
-  </field>
+    <field name="authentication" label="Authentication" category="authentication" value-type="string" editable="false" default-value="oauth" />
 ```
-
-In your connectionBuilder.js file you need to use your DB specific logic to handle how to pass in OAuth attributes. Example:
+## Multiple Authentication Options
+```xml
+    <field name="authentication" label="Authentication" category="authentication" value-type="selection" default-value="auth-user-pass" >
+        <selection-group>
+            <option value="auth-user-pass" label="Username and Password"/>
+            <option value="oauth" label="OAuth"/>
+        </selection-group>
+    </field>
+```
+---
+In your *connectionBuilder.js* file you need to use your DB specific logic to handle how to pass in OAuth attributes. Example:
```

Comment:
```suggestion
In your *propertiesBuilder.js* file you need to use your DB specific logic to handle how to pass in OAuth attributes. Example:
```

Reply:
Secrets need to be handled by the propertiesBuilder...

Reply:
Oh nvmd, I think I see what you are getting at

Reply:
But it needs to be connectionProperties.js